tests/avro/test_cached_client.py

```diff
@@ -125,3 +125,19 @@ def test_multi_register(self):
 
     def hash_func(self):
         return hash(str(self))
+
+    def test_cert_no_key(self):
+        with self.assertRaises(ValueError):
+            self.client = CachedSchemaRegistryClient(url='https://127.0.0.1:8081',
```

Comment:
If we're unlucky the schema registry client will connect to this host and port, which we don't want for a unit test.
Not sure there is a good alternative, but maybe we should use port 65534 instead that is much less likely to be bound.